### PR TITLE
Add instructions for using NGINX dynamic modules and small doc improvements

### PR DESCRIPTION
**Pull request by Gehock**
# Description

This pull request adds instructions about using NGINX dynamic modules to the deployment documentation, as well as improves documentation in other parts.

## Testing

These changes are a result of the Aalto A+ exam server deployment and are currently used in production.

## Is it [Done](https://wiki.aalto.fi/display/EDIT/Definition+of+Done)?

- [x] I (developer) have tested the changes manually
- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly

### CODE REVIEW

File: doc/DEPLOYMENT.md [hunk 1]
```diff
@@ -372,7 +373,34 @@ This module uses fastcgi and shibboleth scripts to provide similar integration a
...
+          --add-dynamic-module=../nginx-http-shibboleth \
+          --add-dynamic-module=../headers-more-nginx-module \
+          FLAGS_HERE
+        make modules
+        chmod 644 \
+           objs/ngx_http_shibboleth_module.so \
+           objs/ngx_http_headers_more_filter_module.so
+        cp objs/ngx_http_shibboleth_module.so \
+           objs/ngx_http_headers_more_filter_module.so \
+           /usr/lib/nginx/modules/
```
Comment (raphendyr) [inline doc/DEPLOYMENT.md:400]: Can this be `/usr/local/lib/nginx/modules/`?

p.s. in Aalto, could CS IT or Aalto package prebuild modules as a debian package and install that with the nginx on relevant servers?

**Review by Mankro — APPROVED** [review]
Nice! I don't know NGINX magic myself but I trust you do. :)
Were you planning to still change something since you did something just today? I will wait a while before merging.

File: doc/DEPLOYMENT.md (hunk 1 above)
Comment (Gehock) [reply doc/DEPLOYMENT.md]: Module path changed to `/usr/local/lib/nginx/modules`. Technically the module could be prebuilt as a debian package but I'd say that setting up shibboleth is such a rare case that it's not worth the effort

**Review by Mankro — APPROVED** [review]
Excellent! Thank you so much!

**Pull request merged**